plugins/modules/vmware_guest_storage_policy.py

```diff
@@ -0,0 +1,458 @@
+            pol_obj = self.SearchStorageProfileByName(pm, policy)
+
+            if not pol_obj:
+                self.module.fail_json(msg="Unable to find storage policy `%s' "
+                                          "for vm_home" % policy)
+
+            if not self.CheckAssociatedStorageProfile(pm, pmRef, policy):
+                # Existing policy is different than requested. Set, wait for
+                # task success, and exit.
+                if not self.module.check_mode:
+                    task = self.SetVMHomeStorageProfile(vm_obj, pol_obj)
+                    wait_for_task(task)  # will raise an Exception on failure
+                result['changed'] = True
+                result['changed_policies']['vm_home'] = policy
+
+        #
+        # DISKS
+        #
+        if disks is None:
+            disks = list()
+        # Check the requested disks[] information is sane or fail by looking up
+        # and storing the object(s) in a new dict.
+        disks_objs = dict()  # {unit_number: {disk: <obj>, policy: <obj>}}
+        for disk in disks:
+            policy = str(disk['policy'])
+            unit_number = int(disk['unit_number'])
+            disk_obj = self.GetVirtualDiskObj(vm_obj, unit_number)
+            pol_obj = self.SearchStorageProfileByName(pm, policy)
+            if not (disk_obj or policy):
+                self.module.fail_json(msg="Unable to find storage policy `%s' "
+                                          "for disk %s" % (policy, disk))
+            disks_objs[unit_number] = dict(disk=disk_obj, policy=pol_obj)
+
+        # All requested profiles are valid. Iterate through each disk and set
+        # accordingly.
+        for disk in disks:
+            policy = str(disk['policy'])
+            unit_number = int(disk['unit_number'])
+            disk_obj = disks_objs[unit_number]['disk']
+            pol_obj = disks_objs[unit_number]['policy']
+            pmObjectType = pbm.ServerObjectRef.ObjectType("virtualDiskId")
+            pmRef = pbm.ServerObjectRef(key="%s:%s"
+                                            % (vm_obj._moId, disk_obj.key),
+                                        objectType=pmObjectType)
+
+            if not self.CheckAssociatedStorageProfile(pm, pmRef, policy):
+                # Existing policy is different than requested. Set, wait for
+                # task success, and exit.
+                if not self.module.check_mode:
+                    task = self.SetVMDiskStorageProfile(vm_obj, unit_number,
+                                                        pol_obj)
+                    wait_for_task(task)
+                result['changed'] = True
+                result['changed_policies']['disk'].append(disk)
+
+        #
+        # END
+        #
+        # Check our results and exit.
+        if result['changed']:
+            result['msg'] = success_msg
+        self.module.exit_json(**result)
+
+
+def run_module():
```

Comment:
There is a module called [`vmware_vm_storage_policy`](https://github.com/ansible-collections/community.vmware/blob/main/plugins/modules/vmware_vm_storage_policy.py). Can we come with some other name so that it is less confusing to the users?

Reply:
What do you suggest? I was under the impression vmware_guest_* was meant for all guest related actions, which this is. Seems to me vmware_vm_storage_policy should be renamed vmware_host_storage_policy...

Reply:
Are we OK with this name or do I still need to change it? I need suggestions if not please.

Reply:
@sky-joker @goneri @mariolenz What do you think about this?

Reply:
As I understand, the vmware_vm_storage_policy module is for creating a storage policy of VM into vCenter.  
It module doesn't assign a policy to any VM.

The vmware_guest_storage_policy module is for assigning a policy to VM.
So I think it is ok that this module name is vmware_vm_storage_policy.

Reply:
ok